Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeSwitcher.cpp

```diff
@@ -0,0 +1,381 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzToolsFramework/ComponentMode/ComponentModeDelegate.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/ComponentMode/EditorComponentModeBus.h>
+#include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/EditorComponentBase.h>
+#include <AzToolsFramework/ViewportSelection/EditorTransformComponentSelection.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+
+namespace AzToolsFramework::ComponentModeFramework
+
+{
+    // ComponentData constructor to fill in component info
+    ComponentData::ComponentData(AZ::EntityComponentIdPair pairId)
+        : m_pairId(pairId)
+    {
+        AZ::ComponentApplicationBus::BroadcastResult(
+            m_entity, &AZ::ComponentApplicationBus::Events::FindEntity, AZ::EntityId(pairId.GetEntityId()));
+        AZ::ComponentId componentId = pairId.GetComponentId();
+
+        m_component = m_entity->FindComponent(componentId);
+
+        m_componentName = AzToolsFramework::GetFriendlyComponentName(m_component);
+
+        AzToolsFramework::EditorRequestBus::BroadcastResult(
+            m_iconPath,
+            &AzToolsFramework::EditorRequestBus::Events::GetComponentEditorIcon,
+            m_component->GetUnderlyingComponentType(),
+            m_component);
+    }
+
+    ComponentModeSwitcher::ComponentModeSwitcher()
+    {
+        ViewportEditorModeNotificationsBus::Handler::BusConnect(GetEntityContextId());
+        EntityCompositionNotificationBus::Handler::BusConnect();
+        ToolsApplicationNotificationBus::Handler::BusConnect();
+
+
+        // Create the switcher
+        ViewportUi::ViewportUiRequestBus::EventResult(
+            m_switcherId,
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::CreateSwitcher,
+            ViewportUi::Alignment::TopLeft);
+
+        // Initial Transform button
+        ViewportUi::ViewportUiRequestBus::EventResult(
+            m_transformButtonId,
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::CreateSwitcherButton,
+            m_switcherId,
+            "../../../../Assets/Editor/Icons/Components/Transform.svg",
+            "Transform");
+
+        ViewportUi::ViewportUiRequestBus::Event(
+            ViewportUi::DefaultViewportId,
+            &ViewportUi::ViewportUiRequestBus::Events::SetSwitcherActiveButton,
+            m_switcherId,
+            m_transformButtonId);
+
+        // Handler
```

Comment:
nit: This comment is redundant and can be removed - or we should flesh it out more to be something like....

```
// create callback for viewport ui switcher buttons
```